offload/liboffload/src/Helpers.hpp

```diff
@@ -13,20 +13,24 @@
 //===----------------------------------------------------------------------===//
 
 #include "OffloadAPI.h"
+#include "Shared/OffloadError.h"
+#include "llvm/Support/Error.h"
 
 #include <cstring>
 
 template <typename T, typename Assign>
-ol_errc_t getInfoImpl(size_t ParamValueSize, void *ParamValue,
-                      size_t *ParamValueSizeRet, T Value, size_t ValueSize,
-                      Assign &&AssignFunc) {
+llvm::Error getInfoImpl(size_t ParamValueSize, void *ParamValue,
+                        size_t *ParamValueSizeRet, T Value, size_t ValueSize,
+                        Assign &&AssignFunc) {
   if (!ParamValue && !ParamValueSizeRet) {
-    return OL_ERRC_INVALID_NULL_POINTER;
+    return error::createOffloadError(error::ErrorCode::INVALID_NULL_POINTER,
```

Comment:
Could we have an overload of `createOffloadError` that takes a `ol_errc_t`? That way we could use `OL_ERRC_INVALID_NULL_POINTER` here instead of the longer `error::ErrorCode::INVALID_NULL_POINTER`. The same is true of the autogenerated entry point code. It's purely subjective but that would seem tidier to me.

Reply:
There are already three overloads of `createOffloadError`, so this would double it to six, which feels like a lot (unless we do some fancy template shenanigans. I also think that it makes sense for C++ code to only use the C++ enums.

Of course, my views are subjective as well, and I'll add it if I'm outvoted.

Reply:
That's fair, I'm ok with leaving it then, especially if we consistently use the C++ enum